src/index.js

```diff
@@ -1,39 +1,78 @@
 import './style.css';
+import { addTask, deleteTask, editTask } from './externalFunctions.js';
 
+const taskInput1 = document.getElementById('task-input-1');
+const taskInput2 = document.getElementById('task-input-2');
 const todoList = document.getElementById('todo-list');
+const clearCompletedButton = document.getElementById('clear-completed');
 
-class Task {
-  constructor(description, completed = false) {
-    this.description = description;
-    this.completed = completed;
-  }
-}
+let tasks = [];
 
-const tasks = [
-  new Task('Wash the dishes'),
-  new Task('Do some exercise'),
-  new Task('Write a pull request'),
-];
+const saveTasks = () => {
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+};
 
-function renderTasks() {
+const renderTasks = () => {
   todoList.innerHTML = '';
 
   tasks.forEach((task) => {
+    const { completed, description } = task;
     const taskItem = document.createElement('li');
-    const checkbox = document.createElement('input');
-    checkbox.type = 'checkbox';
-    checkbox.checked = task.completed;
+    taskItem.innerHTML = `
+      <input type="checkbox" ${completed ? 'checked' : ''}>
+      <span>${description}</span>
+      <span class="delete-icon" style="font-family: arial; float: right; cursor:pointer;"><b>x</b></span>
+    `;
+    taskItem.className = completed ? 'completed' : '';
+    todoList.appendChild(taskItem);
+
+    const span = taskItem.querySelector('span');
+    span.addEventListener('dblclick', () => {
+      editTask(task, span, renderTasks, saveTasks);
+    });
+
+    const checkbox = taskItem.querySelector('input[type="checkbox"]');
     checkbox.addEventListener('click', () => {
-      task.completed = checkbox.checked;
-      taskItem.classList.toggle('completed', task.completed);
+      task.completed = !completed;
+      taskItem.className = completed ? 'completed' : '';
+      saveTasks();
+    });
+
+    const deleteIcon = taskItem.querySelector('.delete-icon');
+    deleteIcon.addEventListener('click', () => {
+      deleteTask(task, tasks, renderTasks, saveTasks);
     });
-    const description = document.createElement('span');
-    description.textContent = task.description;
-    taskItem.appendChild(checkbox);
-    taskItem.appendChild(description);
-    taskItem.classList.toggle('completed', task.completed);
-    todoList.appendChild(taskItem);
   });
+};
+
+if (localStorage.getItem('tasks')) {
+  tasks = JSON.parse(localStorage.getItem('tasks'));
+  renderTasks();
 }
 
-renderTasks();
+const handleKeyPress = (event) => {
+  if (event.key === 'Enter') {
+    const taskDescription = event.target.value.trim();
+
+    if (taskDescription !== '' && taskDescription !== event.target.defaultValue) {
+      const newTask = {
+        description: taskDescription,
+        completed: false,
+        index: tasks.length + 1,
+      };
+      addTask(newTask, tasks, renderTasks, saveTasks);
+      event.target.value = '';
+    }
+  }
+};
```

Comment:
- Kindly, Adding the todo item and rendering the todo item is not working, Please fix it to make it functional :+1: [Watch the video](https://www.loom.com/share/100464a793a8451b9bc8f1e4ff17fdec)

Reply:
@binhussen My bad.  You are right . Let me fix it. Thanks for your update.